types/DatetimeLit.hpp

```diff
@@ -51,53 +54,70 @@ struct DateLit {
         + 1   // -
         + 2;  // Day
 
+  // Years should be between [-kMaxYear, +kMaxYear] inclusive both end.
+  static constexpr std::int32_t kMaxYear = 99999;
+  static constexpr std::uint8_t kBitsNeededForDay = 5u;
+  static constexpr std::uint8_t kBitsNeededForMonth = 4u;
+
   static DateLit Create(const std::int32_t _year,
                         const std::uint8_t _month,
                         const std::uint8_t _day) {
     DateLit date;
-    date.year = _year;
-    date.month = _month;
-    date.day = _day;
+    // Normalize year by adding kMaxYear value, because we try to
+    // encode signed year value into an unsigned integer.
```

Comment:
Yes, it does. This means that the leap year calculations are indeed based on the original year and not the normalized one, which is correct behavior. Thanks for the clarification!